### PR TITLE
Mark Metrics SDK as Mixed/Stable

### PR DESCRIPTION
**Pull request by reyang**
This moves the Metrics SDK specification to Mixed/Stable:
1. Once the spec is marked as "Stable", **breaking changes are no longer allowed**.
2. Additive changes are allowed.

These are the issues that should be resolved before we could merge this PR, unless the issue is specific about things that are not marked as Stable in this PR (e.g. Exemplar):
https://github.com/open-telemetry/opentelemetry-specification/issues?q=is%3Aissue+is%3Aopen+label%3Aspec%3Ametrics+label%3Aarea%3Asdk+label%3Arelease%3Arequired-for-ga+-label%3Aarea%3Asemantic-conventions+

Project board https://github.com/orgs/open-telemetry/projects/9.

image
<sub>([original link](https://www.monkeyuser.com/2018/release-day/))</sub>

### CODE REVIEW

**Review by MrAlias — COMMENTED** [review]
I am not able to approve this as the Go implementation is still being built.

**Comment by tigrannajaryan** [comment]
> I am not able to approve this as the Go implementation is still being built.

I believe this change says the SDK spec is stable. It does not mean that the SDK implementations are stable.

**Comment by MrAlias** [comment]
> > I am not able to approve this as the Go implementation is still being built.
> 
> I believe this change says the SDK spec is stable. It does not mean that the SDK implementations are stable.

Right, sorry, I wasn't clear. Without an implementation it is hard to say if this specification should be marked stable, specifically from the Go SIGs perspective. There are possibilities that parts of the specification are not implementable in the Go implementation. If that is the case there may need to be modifications that would not be compliant with our stability guarantee.

**Comment by carlosalberto** [comment]
@jmacd you have been prototyping something for Go, right?

**Comment by anuraaga** [comment]
There are a couple of TODOs in `sdk.md` they make the file "feel" not stable - perhaps they should be moved to issues and removed?

**Comment by reyang** [comment]
> There are a couple of TODOs in `sdk.md` they make the file "feel" not stable - perhaps they should be moved to issues and removed?

There are two places where we have `TODO`s, and both of them are post-Stable topics (and have issues tracking them).

It seems to be an _okay_ practice? - we've discussed this in the previous Metrics SIG meeting and actually confirmed that the [Stable tracing spec](https://github.com/open-telemetry/opentelemetry-specification/blob/29e776a160893e155fe9d59e9675524035821920/specification/trace/sdk.md) also has `TODO`s.

**Comment by reyang** [comment]
> > > I am not able to approve this as the Go implementation is still being built.
> > 
> > 
> > I believe this change says the SDK spec is stable. It does not mean that the SDK implementations are stable.
> 
> Right, sorry, I wasn't clear. Without an implementation it is hard to say if this specification should be marked stable, specifically from the Go SIGs perspective. There are possibilities that parts of the specification are not implementable in the Go implementation. If that is the case there may need to be modifications that would not be compliant with our stability guarantee.

We've discussed this in the previous Spec SIG meeting. In order to mark the spec as Stable, the following things are required:
1. All the issues that are tagged as `area:sdk` + `spec:metrics` + `release:required-for-ga` are resolved.
2. The Metrics SIG has a good confidence that the spec won't have breaking changes.
3. There are multiple (e.g. 3) language SIGs which are able to follow the spec and implement the metrics SDK.

These are non-goal for Stable:
1. All languages SIGs could finish the implementation.
2. All the clarification questions are answered. Many of them are nice-to-have and can be post Stable (e.g. `area:allowed-for-ga` or `area:after-ga` or `editorial`).

File: specification/metrics/sdk.md [hunk 1]
```diff
@@ -180,17 +182,19 @@ are the inputs:
     not in the list will be ignored. If not provided, all the attribute keys
     will be used by default (TODO: once the Hint API is available, the default
     behavior should respect the Hint if it is available).
-  * The `extra dimensions` which come from Baggage/Context (optional). If not
-    provided, no extra dimension will be used. Please note that this only
-    applies to [synchronous Instruments](./api.md#synchronous-instrument).
+  * **Status**: [Experimental](../document-status.md) - the `extra dimensions`
```
Comment (reyang) [inline specification/metrics/sdk.md:185]: @anuraaga FYI.

**Review by cijothomas — APPROVED** [review]
LGTM.

**Comment by cijothomas** [comment]
The data model and proto are stable already, so can we also mark the exporters , except Prometheus, (i.e [OTLP](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk_exporters/otlp.md) , stdout, inmemory) as stable as well? (separate topic, but related,  hence mentioning it here.)

**Comment by reyang** [comment]
> The data model and proto are stable already, so can we also mark the exporters , except Prometheus, (i.e [OTLP](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk_exporters/otlp.md) , stdout, inmemory) as stable as well? (separate topic, but related, hence mentioning it here.)

#2175

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
Sounds like a good trade off going forward!

**Review by anuraaga — APPROVED** [review]

**Review by jack-berg — APPROVED** [review]

File: specification/metrics/sdk.md [hunk 2]
```diff
@@ -47,6 +47,8 @@
 
 ## MeterProvider
 
+**Status**: [Stable](../document-status.md)
```
Comment (jmacd) [inline specification/metrics/sdk.md:50]:
```suggestion
**Status**: [Feature-freeze](../document-status.md)
```

File: specification/metrics/sdk.md [hunk 3]
```diff
@@ -554,6 +562,8 @@ measurements using the equivalent of the following naive algorithm:
 
 ## MetricReader
 
+**Status**: [Stable](../document-status.md)
```
Comment (jmacd) [inline specification/metrics/sdk.md:565]:
```suggestion
**Status**: [Feature-freeze](../document-status.md)
```

File: specification/metrics/sdk.md [hunk 4]
```diff
@@ -661,6 +671,8 @@ Configurable parameters:
 
 ## MetricExporter
 
+**Status**: [Stable](../document-status.md)
```
Comment (jmacd) [inline specification/metrics/sdk.md:674]:
```suggestion
**Status**: [Feature-freeze](../document-status.md)
```

**Review by jmacd — APPROVED** [review]
I propose several components remain in Feature-freeze. This topic was discussed by the technical committee today and the consensus is that we believe the specification is practically Stable, but that an uninitiated member of the community could probably not implement it from scratch in a way that conforms with our group understanding.

We expect to do a bit of editorial work on this specification as the next 1 or 2 prototypes are completed, and then we can finish marking this Stable.

**Review by jmacd — CHANGES_REQUESTED** [review]
@open-telemetry/technical-committee as discussed I think we can merge this PR after accepting the three changes proposed by me.

**Comment by jmacd** [comment]
@cijothomas please take a look at my proposal for merging this PR.

**Comment by cijothomas** [comment]
> @cijothomas please take a look at my proposal for merging this PR.

[This](https://github.com/open-telemetry/opentelemetry-specification/pull/2175#issuecomment-983880044) comment was calling out that, the Exporter section of the SDK spec is not stable, so the exporters themselves (OTLP/Stdout/InMemory), should not be marked stable.

**Comment by jmacd** [comment]
> Exporter section of the SDK spec is not stable, so the exporters themselves (OTLP/Stdout/InMemory), should not be marked stable.

I'm not sure I agree -- can't a specific exporter specification be stable even if the general interface specification is not? For components of an SDK I see this dependency being real (i.e., an SDK exporter can't be stable of other components it depends on are not stable), but for a specification I don't see a hard connection there. 

E.g., the stderr exporter says the "StderrExporter is an Exporter". This is true and stable and will not change even if we change the specification for Exporter, right?

File: specification/metrics/sdk.md [hunk 5]
```diff
@@ -180,17 +182,19 @@ are the inputs:
     not in the list will be ignored. If not provided, all the attribute keys
     will be used by default (TODO: once the Hint API is available, the default
     behavior should respect the Hint if it is available).
-  * The `extra dimensions` which come from Baggage/Context (optional). If not
-    provided, no extra dimension will be used. Please note that this only
-    applies to [synchronous Instruments](./api.md#synchronous-instrument).
+  * **Status**: [Feature-freeze](../document-status.md) - the `extra dimensions`
```
Comment (jmacd) [inline specification/metrics/sdk.md:185]:
```suggestion
  * the `extra dimensions`
```

File: specification/metrics/sdk.md [hunk 6]
```diff
@@ -180,17 +182,19 @@ are the inputs:
...
+    Instruments](./api.md#synchronous-instrument).
   * The `aggregation` (optional) to be used. If not provided, the SDK MUST
     apply a [default aggregation](#default-aggregation). If the aggregation
     outputs metric points that use aggregation temporality (e.g. Histogram,
     Sum), the SDK SHOULD handle the aggregation temporality based on the
     temporality of each [MetricReader](#metricreader) instance.
-  * The `exemplar_reservoir` (optional) to use for storing exemplars.
-    This should be a factory or callback similar to aggregation which allows
-    different reservoirs to be chosen by the aggregation.
+  * **Status**: [Feature-freeze](../document-status.md) - the
```
Comment (jmacd) [inline specification/metrics/sdk.md:194]:
```suggestion
  * the
```

**Review by jmacd — CHANGES_REQUESTED** [review]
With the suggestion to leave MeterProvider in Feature-freeze, these two sections are redundant.

**Pull request closed without merging**